utils/client/src/precompiles/mod.rs

```diff
@@ -51,37 +52,103 @@ const PRECOMPILES: &[(PrecompileWithAddress, PrecompileWithAddress)] = &[
     ),
 ];
 
-// Source: https://github.com/anton-rs/kona/blob/main/bin/client/src/fault/handler/mod.rs#L20-L42
-pub fn zkvm_handle_register<F, H>(handler: &mut EvmHandler<'_, (), &mut State<&mut TrieDB<F, H>>>)
+/// The ZKVM-cycle-tracking precompiles.
+#[derive(Debug)]
+pub struct OpZkvmPrecompiles {
```

Comment:
Why do we need this? Feels like a lot of code duplication?

Reply:
@ratankaliani Below in line 78, we're overrinding precompiles so that we can track cycles.
```
        precompiles_owned.extend(PRECOMPILES.iter().map(|p| p.1.clone()).take(1));
```